src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            var blockSyntaxNode = methodDeclarationSyntax.Body;
+            expressionSyntax = null;
+            if (blockSyntaxNode != null)
+            {
+                // 1. If it is an ordinary method with block
+                var blockStatements = blockSyntaxNode.Statements;
+                if (blockStatements.Count == 1 && CanStatementBeInlined(blockStatements[0]))
+                {
+                    expressionSyntax = GetExpressionFromStatementSyntaxNode(blockStatements[0]);
+                    return true;
+                }
+            }
+            else
+            {
+                // 2. If it is an Arrow Expression
+                var arrowExpressionNode = methodDeclarationSyntax.ExpressionBody;
+                if (arrowExpressionNode != null)
+                {
+                    expressionSyntax = arrowExpressionNode.Expression;
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static ExpressionSyntax GetExpressionFromStatementSyntaxNode(StatementSyntax statementSyntax)
+            => statementSyntax switch
+            {
+                // Check has been done before to make sure the argument is ReturnStatementSyntax or ExpressionStatementSyntax
+                // and their expression is not null
+                ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression!,
+                ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                _ => throw ExceptionUtilities.Unreachable
+            };
+
+        protected override bool IsSingleStatementOrExpressionMethod(MethodDeclarationSyntax calleeMethodDeclarationSyntaxNode)
+            => TryGetSingleStatementOrExpressionMethod(calleeMethodDeclarationSyntaxNode, out _);
+
+        protected override ExpressionSyntax GetInlineStatement(MethodDeclarationSyntax calleeMethodDeclarationSyntaxNode)
+        {
+            if (TryGetSingleStatementOrExpressionMethod(calleeMethodDeclarationSyntaxNode, out var expressionSyntax))
+            {
+                return expressionSyntax;
+            }
+
+            // Check has been done before to make sure it will not hit here.
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        protected override SyntaxNode? GetEnclosingMethod(SyntaxNode syntaxNode)
+        {
+            for (var node = syntaxNode; node != null; node = node.Parent)
+            {
+                if (node.IsKind(SyntaxKind.MethodDeclaration)
+                    || node.IsKind(SyntaxKind.LocalFunctionStatement)
+                    || node is LambdaExpressionSyntax)
+                {
+                    return node;
+                }
+            }
+
+            return null;
+        }
+
+        protected override SyntaxNode GenerateTypeSyntax(ITypeSymbol symbol)
+            => symbol.GenerateTypeSyntax(allowVar: false);
```

Comment:
why is `var` not allowed?  needs comments.  what if this is the user style?

Reply:
Emm, I think we might use the config from editor config maybe. So it might be better to just read from https://github.com/dotnet/roslyn/blob/1bb0652c3fa646f9aff45ae0372a86e6416ff9be/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs#L30 ?

Reply:
if you just call .GenerateTypeSyntax then it will use `var` if htat matches hte user preference.